tests/test_algorithm.py

```diff
@@ -428,6 +429,55 @@ def test_future_chain(self):
         with self.assertRaises(RootSymbolNotFound):
             algo.future_chain('')
 
+        # Check that invalid dates raise UnsupportedDatetimeFormat
+        with self.assertRaises(UnsupportedDatetimeFormat):
+            algo.future_chain('CL', 'my_finger_slipped')
+
+        with self.assertRaises(UnsupportedDatetimeFormat):
+            algo.future_chain('CL', '2015-09-')
+
+    def test_set_symbol_lookup_date(self):
+        """
+        Test the set_symbol_lookup_date API method.
+        """
+        # Note we start sid enumeration at i+3 so as not to
+        # collide with sids [1, 2] added in the setUp() method.
+        dates = pd.date_range('2013-01-01', freq='2D', periods=2, tz='UTC')
+        # Create two assets with the same symbol but different
+        # non-overlapping date ranges.
+        metadata = pd.DataFrame.from_records(
```

Comment:
@jfkirk As mentioned, this will require refactoring when we merge the asset-db-writer-update branch, but should only involve moving the writing of this metadata into the `setUpClass` method and modifying the call to `TradingAlgorithm`